### ISSUE TITLE
Feature request: wait for all items in queue to finish

### ISSUE THREAD

**Issue by dobesv** [promise-queue/promise-queue#23]
When running a script that runs jobs in parallel, it would be nice to have a promise we could wait on that resolves when all items in the queue are complete.

The current workaround would look like this:

```
while(queue.getQueueLength() + queue.getPendingLength() > 0) await Bluebird.delay(100);
```

If the feature were built into the library a more elegant solution should be possible.

**Comment by ScottA38** [comment]
@dobesv Sorry if I missed the point here, I am by no means a JS programmer, but isn't there some cause for you to use [promise.all()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)?
I am geniuinely curious because as stated, not really an advanced programmer

**Comment by dobesv** [comment]
@ScottA38 

For my use case I have a script that is running and adding jobs to the queue as it goes.

If the queue reaches a certain length while it is working, it stops adding jobs and waits for the queue to drain.

At the end of the script, it waits for all jobs to finish before it exits.

This function would at least help with the last part, where I'm just waiting for all jobs to finish before quitting.

`Promise.all()` is quite as suitable for this as it doesn't work well with the middle part were jobs are being queued with some concurrency limit.

I'm actually using `p-queue` currently for this instead of `promise-queue`, as it has more helpful features along these lines.
